PWGHF/D2H/Tasks/taskCharmResoReduced.cxx

```diff
@@ -0,0 +1,264 @@
+  ConfigurableAxis multPoolBins{"multPoolBins", {VARIABLE_WIDTH, 0., 45., 60., 75., 95, 250}, "event multiplicity pools (PV contributors for now)"};
+  ConfigurableAxis zPoolBins{"zPoolBins", {VARIABLE_WIDTH, -10.0, -4, -1, 1, 4, 10.0}, "z vertex position pools"};
+  // ConfigurableAxis bzPoolBins{"bzPoolBins", {2, -10, 10}, "Bz of collision"};
+
+  using reducedResoWithMl = soa::Join<aod::HfCandCharmReso, aod::HfCharmResoMLs>;
+  SliceCache cache;
+  Preslice<reducedResoWithMl> resoPerCollision = aod::hf_track_index_reduced::hfRedCollisionId;
+
+  // Histogram Registry
+  HistogramRegistry registry;
+
+  // init
+  void init(InitContext&)
+  {
+    const AxisSpec axisPt{configAxisPt, "#it{p}_{T} (GeV/#it{c})"};
+    const AxisSpec axisPtProng0{configAxisPtProng0, "#it{p}_{T} (GeV/#it{c})"};
+    const AxisSpec axisPtProng1{configAxisPtProng1, "#it{p}_{T} (GeV/#it{c})"};
+    const AxisSpec axisInvMassReso{configAxisInvMassReso, "inv. mass (DV_{0}) (GeV/#it{c}^{2})"};
+    const AxisSpec axisInvMassProng0{configAxisInvMassProng0, "inv. mass (D) (GeV/#it{c}^{2})"};
+    const AxisSpec axisInvMassProng1{configAxisInvMassProng1, "inv. mass ({V}_{0}) (GeV/#it{c}^{2})"};
+    const AxisSpec axisCosThetaStar{configAxisCosThetaStar, "cos(#vartheta*)"};
+    const AxisSpec axisBkgBdtScore{configAxisBkgBdtScore, "Bkg BDT Score"};
+    const AxisSpec axisNonPromptBdtScore{configAxisNonPromptBdtScore, "Non Prompt BDT Score"};
+    registry.add("hMass", "Charm resonance candidates inv. mass", {HistType::kTH1F, {axisInvMassReso}});
+    registry.add("hMassProng0", "D daughters inv. mass", {HistType::kTH1F, {axisInvMassProng0}});
+    registry.add("hMassProng1", "V0 daughter inv. mass", {HistType::kTH1F, {axisInvMassProng1}});
+    registry.add("hPt", "Charm resonance candidates pT", {HistType::kTH1F, {axisPt}});
+    registry.add("hPtProng0", "D daughters pT", {HistType::kTH1F, {axisPtProng0}});
+    registry.add("hPtProng1", "V0 daughter pT", {HistType::kTH1F, {axisPtProng1}});
+    registry.add("hNPvCont", "Collision number of PV contributors ; N contrib ; entries", {HistType::kTH1F, {{100, 0, 250}}});
+    registry.add("hZvert", "Collision Z Vtx ; z PV [cm] ; entries", {HistType::kTH1F, {{120, -12., 12.}}});
+    registry.add("hBz", "Collision Bz ; Bz [T] ; entries", {HistType::kTH1F, {{20, -10., 10.}}});
+    if (doprocessDs1Data || doprocessDs1DataME || doprocessDs2StarData || doprocessDs2StarDataME) {
+      registry.add("hSparseMl", "THn for production studies with cosThStar and BDT scores", HistType::kTHnSparseF, {axisPt, axisPtProng0, axisPtProng1, axisInvMassReso, axisInvMassProng0, axisInvMassProng1, axisCosThetaStar, axisBkgBdtScore, axisNonPromptBdtScore});
+    } else {
+      registry.add("hSparse", "THn for production studies with cosThStar", HistType::kTHnSparseF, {axisPt, axisPtProng0, axisPtProng1, axisInvMassReso, axisInvMassProng0, axisInvMassProng1, axisCosThetaStar});
+    }
```

Comment:
I would give the same name to these histograms (to have always the same in the output)
It is also not clear to me why you have 2 versions, given that there is no process function that doesn't have ML

Reply:
You're right, I started with 2 versions just to be consistent with the rest of the workflow but forgot to write process functions for the workflow wihtout Ml (which will never be used). If it's ok with you I would keep only the Ml workflow to keep down the number of process functions